mailpoet/lib/Cron/Workers/SendingQueue/SendingQueue.php

```diff
@@ -272,12 +274,21 @@ public function processQueue($queue, $newsletter, $subscribers, $timer) {
     $unsubscribeUrls = [];
     $statistics = [];
     $metas = [];
+
+    $newsletterEntity = $this->newslettersRepository->findOneById($newsletter->id);
+
     foreach ($subscribers as $subscriber) {
+      $subscriberEntity = $this->subscribersRepository->findOneById($subscriber->id);
+
+      if (!$subscriberEntity instanceof SubscriberEntity) {
+        continue;
+      }
+
       // render shortcodes and replace subscriber data in tracked links
       $preparedNewsletters[] =
         $this->newsletterTask->prepareNewsletterForSending(
-          $newsletter,
-          $subscriber,
+          $newsletterEntity,
```

Comment:
I was wondering how this got through PHPStan because, if I am not wrong, the newsletter entity can be null here.
In the end, I found that the `newsletterTask` property is not annotated.

Reply:
Good catch. I documented the properties and add an if to workaround the PHPStan error. The if is inside a foreach which is not really necessary but it was the easiest way that I found and it shouldn't be a big problem as far as I can see as we will revisit this soon as part of another Doctrine ticket.